spec/inputs/rabbitmq_spec.rb

```diff
@@ -166,21 +166,73 @@ def spawn_and_wait(instance)
   end
 
   describe "receiving a message with a queue specified" do
-    let(:queue_name) { "foo_queue" }
+    let(:queue_name) { "logstash-input-rabbitmq-#{rand(0xFFFFFFFF)}" }
     let(:config) { super.merge("queue" => queue_name) }
+    let(:queue) { test_channel.queue(queue_name, :auto_delete => true) }
 
     it "should process the message" do
       message = "Foo Message"
-      q = test_channel.queue(queue_name)
-      q.publish(message)
+      queue.publish(message)
 
       event = output_queue.pop
       expect(event["message"]).to eql(message)
     end
+
+    it "should save message properties into a @metadata field" do
+      epoch = 1454789198
+      # Don't test every single property but select a few with
+      # different characteristics to get sufficient coverage.
+      queue.publish("",
```

Comment:
My preference is that these setup-style activities belong in a `before` block. Example:

```
context "when message properties are available" do
  before do
    epoch = ...
    queue.publish(...)
  end

  let(:event) { output_queue.pop }

  it "should save these properties to @metadata field" do
    expect(event).to include("@metadata")
    ...
 end
end
```

Bonus that this lets you refactor to use shared examples so you can assert that all common behavior exists for all tests (like event has `@metadata` or something)